### PR TITLE
[ML-5699] Automated release script - publish artifact

### PR DESCRIPTION
**Pull request by MrBago**
This PR adds a script to make the process of publishing a spark-deep-learning releases faster and more automated.

### CODE REVIEW

File: dev/release.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    while response not in ("y", "yes", "n", "no"):
+        response = input(prompt).lower()
+    return response in ("y", "yes")
+
+
+@click.command()
+@click.argument("release_version", type=str)
+@click.argument("next_version", type=str)
+@click.option("--local", is_flag=True)
+@click.option("--skip-tests", is_flag=True)
```
[the remark this answers is not in the document]
Comment (MrBago) [reply dev/release.py]: I switched them to use a hyphen like the other variables, but click actually converts mandatory arguments to use an underscore, even though I used a hyphen the help looks like this:

```
Usage: release.py [OPTIONS] RELEASE_VERSION NEXT_VERSION

Options:
  --publish-to TEXT  Where to publish artifact, one of: ['local', 'spark-
                     package', 'm2']
  --no-prompt        Automated mode with no user prompts.
  --help             Show this message and exit.
```

File: dev/release.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    response = None
+    while response not in ("y", "yes", "n", "no"):
+        response = input(prompt).lower()
+    return response in ("y", "yes")
+
+
+@click.command()
+@click.argument("release_version", type=str)
+@click.argument("next_version", type=str)
+@click.option("--local", is_flag=True)
```
[the remark this answers is not in the document]
Comment (MrBago) [reply dev/release.py]: I've converted this option to take a string so we can specify multiple publish modes.

**Comment by MrBago** [comment]
This is ready for another pass, I'll follow up with another PR for building and publishing docs.

File: dev/release.py [hunk 3]
```diff
@@ -5,37 +5,45 @@
...
 
 @click.command()
-@click.argument("release_version", type=str)
-@click.argument("next_version", type=str)
-@click.option("--local", is_flag=True)
-@click.option("--skip-tests", is_flag=True)
-def main(release_version, next_version, local, skip_tests):
+@click.argument("release-version", type=str)
+@click.argument("next-version", type=str)
+@click.option("--publish-to", default="spark-package",
```
Comment (jkbradley) [inline dev/release.py]: How about defaulting to "local" for safety?

**Comment by jkbradley** [comment]
LGTM
We don't need python-tests to pass, right?  Feel free to merge it.

**Pull request merged**